jetty-server/src/main/java/org/eclipse/jetty/server/handler/StatisticsHandler.java

```diff
@@ -189,7 +189,7 @@ else if (_wrapWarning.compareAndSet(false, true))
             _dispatchedStats.decrement();
             _dispatchedTimeStats.record(dispatched);
 
-            if (state.isSuspended())
+            if (state.isSuspended() || state.isAsyncStarted())
```

Comment:
@gregw Do you think we still need to keep the `state.isSuspended()` check here. The `state.isAsyncStarted()` doesn't check for the `WAITING` state.

Reply:
@thomasdraebing firstly, yes this is indeed the bug.  We spotted it yesterday and were in process of fixing.

The `isSuspended()` check is not necessary as `isAsyncStarted` in his context will always be for a request that is being handled, hence all we need is the !blocking check that it will do.

@lachlan-roberts are you working on the unit tests for this plus the other things that we wanted changed?  If so, then perhaps pull @thomasdraebing commit into your branch then we can close this one.

Reply:
I rebased my changes ontop of this branch and opened PR #5175.